src/App.tsx

```diff
@@ -1,10 +1,56 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const completedTodosNumber = todos.filter(todo => !todo.completed).length;
+  const [selectedFilter, setSelectedFilter] = useState<string>('All');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [showError, setShowError] = useState<boolean>(false);
```

Comment:
```suggestion
  const [showError, setShowError] = useState(false);
```
redundant pass generic primitive types. in general it is state looks like redundant. use errorMessage instead.